cdk/domino_cdk/provisioners/eks/eks_nodegroup.py

```diff
@@ -1,265 +1,52 @@
-            statements=[
-                iam.PolicyStatement(
-                    actions=[
-                        "autoscaling:DescribeAutoScalingGroups",
-                        "autoscaling:DescribeLaunchConfigurations",
-                        "autoscaling:DescribeTags",
-                        "ec2:DescribeLaunchTemplateVersions",
-                    ],
-                    resources=["*"],
-                ),
-                asg_group_statement,
-            ],
-        )
-
-        if r53_zone_ids:
-            self.route53_policy = iam.ManagedPolicy(
-                self.scope,
-                "route53",
-                managed_policy_name=f"{self.name}-route53",
-                statements=[
-                    iam.PolicyStatement(
-                        actions=["route53:ListHostedZones"],
-                        resources=["*"],
-                    ),
-                    iam.PolicyStatement(
-                        actions=[
-                            "route53:ChangeResourceRecordSets",
-                            "route53:ListResourceRecordSets",
-                        ],
-                        resources=[f"arn:aws:route53:::hostedzone/{zone_id}" for zone_id in r53_zone_ids],
-                    ),
-                ],
-            )
-            cdk.CfnOutput(
-                self.scope,
-                "route53-zone-id-output",
-                value=str(r53_zone_ids),
-            )
-            self.outputs["route53-txt-owner-id"] = cdk.CfnOutput(
-                self.scope,
-                "route53-txt-owner-id",
-                value=f"{self.name}CDK",
-            )
-
-        self.ecr_policy = iam.ManagedPolicy(
-            self.scope,
-            "DominoEcrReadOnly",
-            managed_policy_name=f"{self.name}-DominoEcrRestricted",
-            statements=[
-                iam.PolicyStatement(
-                    effect=iam.Effect.DENY,
-                    actions=["ecr:*"],
-                    conditions={"StringNotEqualsIfExists": {"ecr:ResourceTag/domino-deploy-id": self.name}},
-                    resources=[f"arn:aws:ecr:*:{self.scope.account}:*"],
-                ),
-            ],
-        )
-
-        managed_policies = [
-            s3_policy,
-            self.ecr_policy,
-            self.autoscaler_policy,
-            iam.ManagedPolicy.from_aws_managed_policy_name('AmazonEKSWorkerNodePolicy'),
-            iam.ManagedPolicy.from_aws_managed_policy_name('AmazonEC2ContainerRegistryReadOnly'),
-            iam.ManagedPolicy.from_aws_managed_policy_name('AmazonEKS_CNI_Policy'),
-            iam.ManagedPolicy.from_aws_managed_policy_name('AmazonSSMManagedInstanceCore'),
-        ]
-        if r53_zone_ids:
-            managed_policies.append(self.route53_policy)
+        def provision_nodegroup(nodegroup: EKS.NodegroupBase, prov_func):
+            for name, ng in nodegroup.items():
+                if not ng.machine_image or not ng.machine_image.ami_id:
```

Comment:
I did it in my change but forgot to update the template.

ami_id and user_data become independent values.

we need to eliminate "machine_image" layer and more ami_id and user_data up.